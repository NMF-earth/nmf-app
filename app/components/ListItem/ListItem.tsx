import React from "react";
import { TouchableOpacity } from "react-native";
import { Ionicons } from "@expo/vector-icons";

import { Colors } from "style";

import styles from "./ListItem.styles";
import Text from "../Text";

interface Props {
  title: string;
  onPress: () => void;
}

const ListItem = ({ title, onPress }: Props) => {
  return (
    <TouchableOpacity onPress={onPress} style={styles.container}>
      <Text.Secondary numberOfLines={1} style={styles.text}>
        {title}
      </Text.Secondary>
      <Ionicons
        style={styles.icon}
        name={"ios-arrow-forward"}
        size={26}
        color={Colors.grey40}
      />
    </TouchableOpacity>
  );
};

export default ListItem;
