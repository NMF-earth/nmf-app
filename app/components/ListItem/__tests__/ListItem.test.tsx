import "react-native";
import React from "react";
import { create } from "react-test-renderer";

import ListItem from "../ListItem";

describe("<ListItem />", () => {
  const defaultProps = {
    selected: true,
    title: "France",
    onPress: () => {
      // do nothing.
    },
  };
  const wrapper = create(<ListItem {...defaultProps} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
