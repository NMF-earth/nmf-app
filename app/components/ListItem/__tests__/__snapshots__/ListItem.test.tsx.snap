// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> ListItem renders correctly both topLine and bottomLine 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 18,
      },
      {
        "borderTopColor": "#F0F0F0",
        "borderTopWidth": 1.6,
      },
      {
        "borderBottomColor": "#F0F0F0",
        "borderBottomWidth": 1.6,
      },
    ]
  }
>
  <Text.Secondary
    numberOfLines={1}
  >
    France
  </Text.Secondary>
  <Ionicons
    color="#878787"
    name="ios-chevron-forward-outline"
    size={14}
  />
</TouchableOpacity>
`;

exports[`<ListItem /> ListItem renders correctly wtesth bottomLine 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 18,
      },
      {
        "borderBottomColor": "#F0F0F0",
        "borderBottomWidth": 1.6,
      },
    ]
  }
>
  <Text.Secondary
    numberOfLines={1}
  >
    France
  </Text.Secondary>
  <Ionicons
    color="#878787"
    name="ios-chevron-forward-outline"
    size={14}
  />
</TouchableOpacity>
`;

exports[`<ListItem /> ListItem renders correctly wtesth topLine 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 18,
      },
      {
        "borderTopColor": "#F0F0F0",
        "borderTopWidth": 1.6,
      },
    ]
  }
>
  <Text.Secondary
    numberOfLines={1}
  >
    France
  </Text.Secondary>
  <Ionicons
    color="#878787"
    name="ios-chevron-forward-outline"
    size={14}
  />
</TouchableOpacity>
`;

exports[`<ListItem /> render 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 18,
      },
    ]
  }
>
  <Text.Secondary
    numberOfLines={1}
  >
    France
  </Text.Secondary>
  <Ionicons
    color="#878787"
    name="ios-chevron-forward-outline"
    size={14}
  />
</TouchableOpacity>
`;
